internal/downloader/downloader_test.go

```diff
@@ -61,20 +65,52 @@ func TestDownloader_Download(t *testing.T) {
 			source: "http://example.com",
 			err:    errors.New("attempting to download from insecure source: http://example.com"),
 		},
+		{
+			name:        "experimental download",
+			dest:        "dir",
+			source:      "example.com/repo.git",
+			useGoGather: "1",
+		},
+		{
+			name:        "experimental download with error",
+			dest:        "dir",
+			source:      "example.com/repo.git",
+			useGoGather: "1",
+			err:         errors.New("gather error"),
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			d := mockDownloader{}
 			ctx := WithDownloadImpl(context.TODO(), &d)
-			d.On("Download", ctx, tt.dest, []string{tt.source}).Return(tt.err)
+
+			originalGatherFunction := gatherFunc
+
+			if tt.useGoGather == "1" {
+				t.Setenv("USEGOGATHER", "1")
+				gatherFunc = func(_ context.Context, _ string, _ string) (metadata.Metadata, error) {
```

Comment:
I might be missing something, but I'm confused about how it downloads from `example.com/repo.git` if it's not using `mockDownloader{}`

Reply:
Oh, is gatherFunc here being set to a mocked function, but different to mock downloader?

Reply:
yes